### ISSUE TITLE
Rewrite chunk uploads

### ISSUE THREAD

**Issue by Pitu** [WeebDev/chibisafe#243]
With the method bobby pointed out it would be possible to use a stream writer to save the files directly to disk to minimize ram usage, specially great for big file uploads. The only caveat which needs testing to see if it affects speed/performance is that it needs parallel chunk uploads disabled.

https://github.com/BobbyWibowo/lolisafe/blob/safe.fiery.me/controllers/multerStorageController.js

**Comment by BobbyWibowo** [comment]
I'd like to add that I believe the biggest advantage of this approach is, the ability to hash the uploads in parallel to them being uploaded and written to disk.

A summary of https://github.com/BobbyWibowo/lolisafe/issues/191:

A user previously claimed that uploading +20GBs file would take his machine ~1 minute and 30 seconds in "rebuilding chunks" stage. At first, I thought it just didn't make much sense for it to take that long to simply join the chunks back into a single file. So after some back and forth we concluded the process-intensive task was simply the hashing (*I can only imagine how slow that'd be on a machine as potato as my server is*).
A fellow dev then kindly suggested blake3 as an alternative, and to look into stream-based approach. That's how I came up with keeping the write stream alive stuff, to hash the stream in parallel, AND to switch to blake3.

The previous user later claimed that his machine returned the link to his 20GB+ upload almost instantly as it finished uploading, which indeed made sense, as by then the service only had to move the upload to its proper location and store its meta in DB.

*Talking about potato servers, this approach will also make user experience with potato servers snappier, lmao*
